ocaml/testsuite/tests/typing-layouts/basics.ml

```diff
@@ -1463,7 +1714,399 @@ Line 1, characters 28-40:
+       Their parameters differ:
+       The type ('a : any) is not equal to the type ('a0 : value)
+       because their layouts are different.
+|}]
+(* CR layouts: This one should be fine to accept *)
+
+(*****************************************************)
+(* Test 38: Ensure Univar unification checks layouts *)
+
+let poly : ('a. 'a -> 'a) -> int * bool =
+  fun (id : ('a : immediate). 'a -> 'a) -> id 3, id true
+
+[%%expect{|
+Line 2, characters 7-38:
+2 |   fun (id : ('a : immediate). 'a -> 'a) -> id 3, id true
+           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: This pattern matches values of type ('a : immediate). 'a -> 'a
+       but a pattern was expected which matches values of type 'a. 'a -> 'a
+       Type 'a is not compatible with type 'a0
+|}]
+(* CR layouts: This one should be fine to accept *)
+
+type ('a : any) foo = 'a
+type ('a : any) bar
+
+let f (x : < foo : ('a : float64) . 'a foo bar >)
+  : < foo : 'a . 'a foo bar > = x
+
+[%%expect{|
+type ('a : any) foo = 'a
+type ('a : any) bar
+Line 5, characters 32-33:
+5 |   : < foo : 'a . 'a foo bar > = x
+                                    ^
+Error: This expression has type < foo : ('a : float64). 'a foo bar >
+       but an expression was expected of type < foo : 'a. 'a foo bar >
+       Type 'a foo = 'a is not compatible with type 'a0 foo = 'a0
+       Types for method foo are incompatible
+|}]
+
+(*************************************************************)
+(* Test 39: Inference of functions that don't bind arguments *)
+
+let rec f () : 'a -> 'a = f ()
+
+[%%expect{|
+val f : unit -> 'a -> 'a = <fun>
+|}]
+
+module M = struct
+  let rec f () : 'a -> 'a = f ()
+end
+
+[%%expect{|
+module M : sig val f : unit -> 'a -> 'a end
+|}]
+
+let rec f () : 'a -> 'a = f ()
+let g : ('a : any). unit -> 'a -> 'a = f
+
+[%%expect{|
+val f : unit -> 'a -> 'a = <fun>
+Line 2, characters 39-40:
+2 | let g : ('a : any). unit -> 'a -> 'a = f
+                                           ^
+Error: This definition has type unit -> 'b -> 'b which is less general than
+         ('a : any). unit -> 'a -> 'a
+       'a has layout any, which is not a sublayout of value.
+|}]
+
+module M : sig
+  val f : ('a : any). unit -> 'a -> 'a
+end = struct
+  let rec f () : 'a -> 'a = f ()
+end
+
+[%%expect{|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   let rec f () : 'a -> 'a = f ()
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig val f : unit -> ('a -> 'a) end
+       is not included in
+         sig val f : ('a : any). unit -> 'a -> 'a end
+       Values do not match:
+         val f : unit -> ('a -> 'a)
+       is not included in
+         val f : ('a : any). unit -> 'a -> 'a
+       The type unit -> ('a -> 'a) is not compatible with the type
+         unit -> 'b -> 'b
+       'a has layout any, which is not representable.
+|}]
+
+let rec f : ('a : any). unit -> 'a -> 'a = fun () -> f ()
+
+[%%expect{|
+val f : ('a : any). unit -> 'a -> 'a = <fun>
```

Comment:
If _this_ didn't work, we'd have a major problem. But it does work, and it means we can actually define functions that deal with `any`. Good. (Just Resolve -- no action necessary.)